src/producer/mod.rs

```diff
@@ -368,3 +387,30 @@ where
     /// [`RDKafkaErrorCode::PurgeQueue`]: crate::error::RDKafkaErrorCode::PurgeQueue
     fn abort_transaction<T: Into<Timeout>>(&self, timeout: T) -> KafkaResult<()>;
 }
+
+bitflags::bitflags! {
+    /// Flags to provide to [`Producer::purge`] to parametrize the purge behavior
+    ///
+    /// This struct is generated via [`bitflags`], so you may find some relevant documentation there.
+    ///
+    /// # Example
+    /// To purge both queued messages and in-flight messages:
+    /// ```
+    /// # use rdkafka::producer::PurgeFlags;
+    /// let flags = PurgeFlags::QUEUE | PurgeFlags::INFLIGHT;
+    /// ```
+    pub struct PurgeFlags: i32 {
+        /// Purge messages in internal queues. This does not purge inflight messages.
+        const QUEUE = rdkafka_sys::RD_KAFKA_PURGE_F_QUEUE;
+        /// Purge messages in-flight to or from the broker.
+        /// Purging these messages will void any future acknowledgements from the
+        /// broker, making it impossible for the application to know if these
+        /// messages were successfully delivered or not.
+        /// Retrying these messages may lead to duplicates.
+        ///
+        /// This does not purge messages in internal queues.
+        const INFLIGHT = rdkafka_sys::RD_KAFKA_PURGE_F_INFLIGHT;
+        /// Don't wait for background thread queue purging to finish.
+        const NON_BLOCKING = rdkafka_sys::RD_KAFKA_PURGE_F_NON_BLOCKING;
+    }
+}
```

Comment:
I feel like it would be Rust-ier to use a `PurgeConfig` struct here with `queue(bool)`, `in_flight(bool)`, and `non_blocking(bool)` methods. No particular need to save the two bytes. Also avoids the dependency on the `bitflags` crate, which idk someone will probably care about. (We get a lot of PRs to minimize crate dependencies.)

Reply:
Done in 37e318097d8238eff430af5c301c5972bd35c935